lib/publiccloud/utils.pm

```diff
@@ -210,4 +211,29 @@ sub define_secret_variable {
     script_run("set +a");
 }
 
+# Get credentials from the Public Cloud micro service, which requires user
+# and password. The resulting json will be stored in a file.
+sub get_credentials {
+    my $url = get_var('PUBLIC_CLOUD_CREDENTIALS_URL');
+    my $user = get_required_var('_SECRET_PUBLIC_CLOUD_CREDENTIALS_USER');
+    my $pwd = get_required_var('_SECRET_PUBLIC_CLOUD_CREDENTIALS_PASS');
+    my $credentials_json = '/root/credentials.json';
+    # Store user/pwd in a file to be used by the curl command
+    script_run('read -s creds', 0);
+    type_password("-u $user:$pwd\n");
+    assert_script_run('echo $creds > curl_creds');
+
+    # Redirect output to a file not to be visible in the logs
+    assert_script_run("curl -K curl_creds $url > $credentials_json");
```

Comment:
```suggestion
    assert_script_run("curl -K curl_creds $url -o $credentials_json");
```

Reply:
No. Let curl print the metrics and the error in case of error.

Reply:
I don't have a strong opinion on that point and having the error message might be useful in some points. Let's see what Jose thinks about it but I would not mind in just using `-o` alone.

Reply:
Yes, metrics and error message is useful. Together with `-f`, we can print the output error if failure, and we can see the 404 or 401. So, I would keep it with `-o` only.